HPC console documentation page (#147)
eth-cscs/cscs-docs: docs/services/hpc-console.md

[](){#hpc-console}
# HPC Console

> HCP Console is a web application designed to provide essential functionalities for interacting with [FirecREST API v2 backend][ref-firecrest].

Reviewer: I wonder a bit about the wording here. Is it functionality for interacting with Alps/the system or with firecrest? It seems like from the user's perspective it's functionality for interacting with the system _using_ firecrest?

```suggestion
HPC Console is a web application designed to provide essential functionalities for interacting with Alps through the [FirecREST API][ref-firecrest].
```
?

> It provides users with a modern web-based environment to interact with HPC resources such as clusters, job schedulers, and filesystems.

Reviewer 2: Maybe we can merge both sentences into one, like this:

```
HPC Console is a web application that provides users with a modern web-based environment to interact with HPC resources such as clusters, job schedulers, and filesystems.
```

## Features

> - A modern web interface built on FirecREST API functionalities.

Reviewer 2: I would remove this line, because it's a secondary thing to mention FirecREST as the backend

- A dashboard offering an overview of configured clusters and their statuses.
- A dedicated view for creating and monitoring jobs.
- A file navigator with basic file management functionalities, including upload and download features.

A description of the views and functionalities can be found [here](https://eth-cscs.github.io/firecrest-ui/documentation/).

## Views overview

- Dashboard – shows cluster availability and basic system status.
- Jobs – allows creating, submitting, and monitoring jobs. Users can provide job name, script, and resource requirements.

> - File Navigator – supports browsing directories, uploading, downloading, and deleting files. Editing files inline is not supported.

Reviewer: This again kind of repeats what was above, adding a few new details. Do you think it makes sense to consolidate the text a bit?

## Supported version

Starting early 2025, CSCS has introduced a new version of the API: [FirecREST version 2](https://eth-cscs.github.io/firecrest-v2).

> This web interface is available and runs on the FirecREST API v2 backend, FirecREST API v1 is not supported.

Reviewer: One thing that's not quite clear to me from this document (given the split between "HPC console" and "firecrest UI" links): is this page documentation about the particular _instance_ of firecrest UI hosted at CSCS for CSCS clusters, or is this a description of firecrest UI (with a renaming in flux)? If the former, I'm not sure if a user cares here about which version of firecrest is being used.

However, I may very well be missing some nuance as to why it's useful to mentionn this here.

## Alps platforms

HPC console is currently available on three [Alps platforms][ref-alps-platforms].

| Platform       | URL               | Clusters                                                                 |
|----------------|----------------------|--------------------------------------------------------------------------|

> | HPC Platform   | [https://my.hpcp.cscs.ch/](https://my.hpcp.cscs.ch/) | [Daint](../clusters/daint), [Eiger](../clusters/eiger)             |
> | ML Platform    | [https://my.mlp.cscs.ch/](https://my.mlp.cscs.ch/)  | [Bristen](../clusters/bristen), [Clariden](../clusters/clariden)  |
> | CW Platform    | [https://my.cwp.cscs.ch/](https://my.cwp.cscs.ch/)  | [Santis](../clusters/santis)  |

Reviewer: ```suggestion
| HPC Platform   | [https://my.hpcp.cscs.ch/](https://my.hpcp.cscs.ch/) | [Daint][ref-cluster-daint], [Eiger][ref-cluster-eiger]             |
| ML Platform    | [https://my.mlp.cscs.ch/](https://my.mlp.cscs.ch/)  | [Bristen][ref-clsuter-bristen], [Clariden][ref-cluster-clariden]  |
| CW Platform    | [https://my.cwp.cscs.ch/](https://my.cwp.cscs.ch/)  | [Santis][ref-cluster-santis]  |
```

Additionally, it may be useful to add these links to each of the cluster pages (instead of? or additionally? not quite sure). They already mention jupyterhub and firecrest as "services" available. See e.g. `docs/clusters/eiger.md`.

## Access and prerequisites

- Access requires a valid CSCS account and authentication through the institutional login system.
- Supported browsers: recent versions of Chrome, Firefox, and Safari are recommended.
- Sessions automatically expire after a period of inactivity.

## Feedback

We welcome your feedback to help us improve HPC Console.
If you encounter issues or have suggestions, please let us know via our [Service Desk](https://jira.cscs.ch/plugins/servlet/desk).

## Further information

> * [HPC Console doc](https://eth-cscs.github.io/firecrest-ui/home/)
> * [Github Repo](https://github.com/eth-cscs/firecrest-ui)
> * [FirecREST doc](https://eth-cscs.github.io/firecrest-v2)

Reviewer: ```suggestion
* [HPC Console documentation](https://eth-cscs.github.io/firecrest-ui/home/)
* [Github repository](https://github.com/eth-cscs/firecrest-ui)
* [FirecREST documentation](https://eth-cscs.github.io/firecrest-v2)
```

Reviewer: C.f. my other question about firecrest ui vs hpc console: Should the first link really say `HPC Console` or should it say `Firecrest UI`? It's not clear to me if the latter will be renamed, or if HPC Console is just an instance of the latter.
